src/middlewared/middlewared/plugins/pool.py

```diff
@@ -2765,28 +2765,30 @@ def query(self, filters, options):
         """
         # Optimization for cases in which they can be filtered at zfs.dataset.query
         zfsfilters = []
+        children_filters = []
         filters = filters or []
         if len(filters) == 1 and len(filters[0]) == 3 and list(filters[0][:2]) == ['id', '=']:
             zfsfilters.append(copy.deepcopy(filters[0]))
 
         sys_config = self.middleware.call_sync('systemdataset.config')
         if sys_config['basename']:
-            filters.extend([
+            children_filters.extend([
                 ['id', '!=', sys_config['basename']],
                 ['id', '!^', f'{sys_config["basename"]}/'],
             ])
 
         # top level dataset that stores all things related to gluster config
         # needs to be hidden from local webUI. (This is managed by TrueCommander)
-        filters.extend([
+        children_filters.extend([
             ['id', 'rnin', '.glusterfs'],
         ])
 
         if osc.IS_LINUX:
             k8s_config = self.middleware.call_sync('kubernetes.config')
             if k8s_config['dataset']:
-                filters.append(['id', '!^', f'{k8s_config["dataset"]}/'])
+                children_filters.append(['id', '!^', f'{k8s_config["dataset"]}/'])
```

Comment:
Do we exclude `k8s_config["dataset"]` itself?

Reply:
Nope, we still show it

Reply:
The attachment delegate should handle that case if the user intends to delete the dataset

Reply:
Good point :)